tests/python/pants_test/option/test_options.py

```diff
@@ -910,7 +895,7 @@ def test_deprecated_options_flag(self):
     # Make sure the warnings don't come out for regular options.
     with self.warnings_catcher() as w:
       self._parse('./pants stale --pants-foo stale --still-good')
-      self.assertEqual(0, len(w))
+      self.assertEqual(5, len(w))
```

Comment:
It's unclear why this changed here. I only see options having been removed in this test, so the warning count having gone up is surprising. Especially given the comment above this.

Reply:
Indeed, but we still register 5 test options that have past the `removal_version`, each of which are generating a warning (below) upon registration. Perhaps the comment should be removed with the new behavior?

```
DEPRECATED: option 'global_crufty' in global scope will be removed in version 999.99.9.dev0.
 use a less crufty global option
DEPRECATED: option 'global_crufty_boolean' in global scope will be removed in version 999.99.9.dev0.
 say no to crufty global options
DEPRECATED: option 'global_delayed_but_already_passed_deprecated_option' in global scope will be removed in version 999.99.9.dev0.
DEPRECATED: option 'crufty' in scope 'stale' will be removed in version 999.99.9.dev0.
 use a less crufty stale scoped option
DEPRECATED: option 'crufty_boolean' in scope 'stale' will be removed in version 999.99.9.dev0.
 say no to crufty, stale scoped options
```

Reply:
Would you mind investigating, and thinking about what behaviour you expect here?

Reply:
@stuhood I now understand the intent of `print_warning` from the previous PR. Tho with that implementation [`CodeRemovedError`'s were being raised](https://github.com/cosmicexplorer/pants/blob/2e41010ede17f728843da20a8892bfb2c51860d7/src/python/pants/base/deprecated.py#L191-L207) when a `removal_version` was greater than `VERSION` and `print_warning` was `False`, despite the option not being deprecated.

I've since reverted the tests and moved `print_warning` to suppress `WarningMessage` objects for deprecated options without raising `CodeRemovedError`'s.